cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -195,184 +172,168 @@ async def _batch_async(
+            return self._event_loop.run_until_complete(self._prompt_async(prompt))
 
-        elif isinstance(prompt, str):
-            if not (options is None or isinstance(options, dict)):
-                raise ValueError(
-                    "options must be a single TLMOptions object for single prompt.\n"
-                    "See: https://help.cleanlab.ai/reference/python/trustworthy_language_model/#class-tlmoptions"
-                )
+        return self._batch_prompt(prompt)
 
-            return self._event_loop.run_until_complete(
-                self.prompt_async(
-                    prompt,
-                    cast(Union[None, TLMOptions], options),
-                    retries=retries,
-                )
-            )
+    async def prompt_async(
+        self,
+        prompt: Union[str, Sequence[str]],
+        /,
+    ) -> Union[TLMResponse, List[TLMResponse]]:
+        """
+        (Asynchronously) Get response and trustworthiness score from TLM.
 
-        else:
-            raise ValueError("prompt must be a string or list of strings.")
+        Args:
+            prompt (str | Sequence[str]): prompt (or list of multiple prompts) for the TLM
+        Returns:
+            TLMResponse | List[TLMResponse]: [TLMResponse](#class-tlmresponse) object containing the response and trustworthiness score.
+                    If multiple prompts were provided in a list, then a list of such objects is returned, one for each prompt.
+        """
+        validate_tlm_prompt(prompt)
 
-    async def prompt_async(
+        async with aiohttp.ClientSession() as session:
+            if isinstance(prompt, str):
+                return await self._prompt_async(prompt, session)
+
+            return cast(
+                List[TLMResponse],
+                await self._batch_async([self._prompt_async(p, session) for p in prompt]),
+            )
+
+    async def _prompt_async(
         self,
         prompt: str,
-        options: Optional[TLMOptions] = None,
         client_session: Optional[aiohttp.ClientSession] = None,
-        retries: int = 0,
     ) -> TLMResponse:
         """
-        (Asynchronously) Get response and confidence from TLM.
+        Private asynchronous method to get response and trustworthiness score from TLM.
 
         Args:
             prompt (str): prompt for the TLM
-            options (Optional[TLMOptions]): options to parameterize TLM with. Defaults to None.
-            client_session (Optional[aiohttp.ClientSession]): async HTTP session to use for TLM query. Defaults to None.
-            retries (int): number of retries for TLM query. Defaults to 0.
         Returns:
-            TLMResponse: [TLMResponse](#class-tlmresponse) object containing the response and confidence score
+            TLMResponse: [TLMResponse](#class-tlmresponse) object containing the response and trustworthiness score
         """
+
         async with self._query_semaphore:
             tlm_response = await api.tlm_prompt(
```

Comment:
need to apply timeout here?

Reply:
Added timeout but not here, I don't think we can add it in this helper function bc the batch function calls this helper function and we want a overall timeout there (not on each individual prompt).